core/vm/interpreter.go

```diff
@@ -133,6 +140,9 @@ func (in *Interpreter) Run(contract *Contract, input []byte) (ret []byte, err er
 	)
 	contract.Input = input
 
+	// Reclaim the stack as an int pool when the execution stops
+	defer in.intPool.put(stack.data...)
```

Comment:
Ah, this is not good. The issue is that the parameters are evaluated here already (i.e. stack.data... evaluates into an empty list). https://play.golang.org/p/A5WL8nCKrIs

We need to defer the parameter evaluation too.

```go
defer func() { in.intPool.put(stack.data...) }()
```

https://play.golang.org/p/4ayoTNxgwsY